pkg/events/dependencies/event.go

```diff
@@ -9,20 +9,24 @@ import (
 // EventNode represent an event in the dependencies tree.
 // It should be read-only for other packages, as it is internally managed.
 type EventNode struct {
-	id                 events.ID
-	explicitlySelected bool
-	dependencies       events.Dependencies
+	id                   events.ID
+	explicitlySelected   bool
+	originalDependencies events.Dependencies // The original dependencies from event definition
+	activeDependencies   events.Dependencies // Currently active dependencies (may be a fallback)
+	currentFallbackIndex int                 // Index of current fallback (-1 means using primary dependencies)
 	// There won't be more than a couple of dependents, so a slice is better for
 	// both performance and supporting efficient thread-safe operation in the future
 	dependents []events.ID
 }
 
 func newDependenciesNode(id events.ID, dependencies events.Dependencies, chosenDirectly bool) *EventNode {
 	return &EventNode{
-		id:                 id,
-		explicitlySelected: chosenDirectly,
-		dependencies:       dependencies,
-		dependents:         make([]events.ID, 0),
+		id:                   id,
+		explicitlySelected:   chosenDirectly,
+		originalDependencies: dependencies,
+		activeDependencies:   dependencies,
+		currentFallbackIndex: -1, // -1 means using primary dependencies
```

Comment:
There is some redundancy here.
We should either have activeDependencies or currentFallbackIndex, not both.